### ISSUE TITLE
CI: Maelstrom persist failure in CI

### ISSUE THREAD

**Issue by jkosh44** [MaterializeInc/materialize#14575]
### What version of Materialize are you using?

`main`

### How did you install Materialize?

Docker image

### What is the issue?

"Maelstrom coverage of persist" failed for me in CI. The error was 
> Analysis invalid! (?????? ???

I was unable to download the actual logs. Retrying fixed the failure.

https://buildkite.com/materialize/tests/builds/40747#0182fa0b-c092-473e-97e8-aa069e386236

### Relevant log output

```source
$ docker compose run maelstrom-persist --time-limit=5 --node-count=1 --concurrency=2 --rate=100 -- maelstrom --blob-uri=mem:// --consensus-uri=mem://
INFO [2022-09-01 17:42:42,875] main - jepsen.cli Test options:
 {:args ["maelstrom" "--blob-uri=mem://" "--consensus-uri=mem://"],
 :log-net-send false,
 :node-count 1,
 :max-txn-length 4,
 :concurrency 2,
 :max-writes-per-key 16,
 :leave-db-running? false,
 :logging-json? false,
 :nemesis-interval 10,
 :log-stderr false,
 :ssh
 {:dummy? false,
  :username "root",
  :password "root",
  :strict-host-key-checking false,
  :private-key-path nil},
 :rate 100.0,
 :argv
 ("test"
  "-w"
  "txn-list-append"
  "--bin=/usr/local/bin/persistcli"
  "--time-limit=5"
  "--node-count=1"
  "--concurrency=2"
  "--rate=100"
  "--"
  "maelstrom"
  "--blob-uri=mem://"
  "--consensus-uri=mem://"),
 :nemesis #{},
 :nodes ["n0"],
 :test-count 1,
 :latency {:mean 0, :dist :constant},
 :bin "/usr/local/bin/persistcli",
 :log-net-recv false,
 :time-limit 5,
 :workload :txn-list-append,
 :consistency-models [:strict-serializable],
 :topology :grid}
 
INFO [2022-09-01 17:42:42,899] jepsen test runner - jepsen.core Command line:
lein run test -w txn-list-append --bin=/usr/local/bin/persistcli --time-limit=5 --node-count=1 --concurrency=2 --rate=100 -- maelstrom --blob-uri=mem:// --consensus-uri=mem://
INFO [2022-09-01 17:42:42,964] jepsen test runner - jepsen.core Running test:
{:args ["maelstrom" "--blob-uri=mem://" "--consensus-uri=mem://"]
 :remote
 #jepsen.control.retry.Remote{:remote #jepsen.control.scp.Remote{:cmd-remote #jepsen.control.clj_ssh.Remote{:concurrency-limit 8,
                                                                                                            :conn-spec nil,
                                                                                                            :session nil,
                                                                                                            :semaphore nil},
                                                                 :conn-spec nil},
                              :conn nil}
 :log-net-send false
 :node-count 1
 :max-txn-length 4
 :concurrency 2
 :db
 #object[maelstrom.db$db$reify__16574
         "0x35738a92"
         "maelstrom.db$db$reify__16574@35738a92"]
 :max-writes-per-key 16
 :leave-db-running? false
 :name "txn-list-append"
 :logging-json? false
 :start-time
 #object[org.joda.time.DateTime "0x10ec54c3" "2022-09-01T17:42:42.000Z"]
 :nemesis-interval 10
 :net
 #object[maelstrom.net$jepsen_net$reify__15683
         "0x2005d3e0"
         "maelstrom.net$jepsen_net$reify__15683@2005d3e0"]
 :client
 #object[maelstrom.workload.txn_list_append$client$reify__18428
         "0x575a6540"
         "maelstrom.workload.txn_list_append$client$reify__18428@575a6540"]
 :barrier
 #object[java.util.concurrent.CyclicBarrier
         "0x7e544223"
         "java.util.concurrent.CyclicBarrier@7e544223"]
 :log-stderr false
 :pure-generators true
 :ssh {:dummy? true}
 :rate 100.0
 :checker
 #object[jepsen.checker$compose$reify__9656
         "0xc6c5b66"
         "jepsen.checker$compose$reify__9656@c6c5b66"]
 :argv
 ("test"
  "-w"
  "txn-list-append"
  "--bin=/usr/local/bin/persistcli"
  "--time-limit=5"
  "--node-count=1"
  "--concurrency=2"
  "--rate=100"
  "--"
  "maelstrom"
  "--blob-uri=mem://"
  "--consensus-uri=mem://")
 :nemesis
 (jepsen.nemesis.ReflCompose
  {:fm {:start-partition 0,
        :stop-partition 0,
        :kill 1,
        :start 1,
        :pause 1,
        :resume 1},
   :nemeses [#unprintable "jepsen.nemesis.combined$partition_nemesis$reify__16846@2f803d8"
             #unprintable "jepsen.nemesis.combined$db_nemesis$reify__16827@68c06a2"]})
 :nodes ["n0"]
 :test-count 1
 :latency {:mean 0, :dist :constant}
 :bin "/usr/local/bin/persistcli"
 :generator
 (jepsen.generator.TimeLimit
  {:limit 5000000000,
   :cutoff nil,
   :gen (jepsen.generator.Any
         {:gens [(jepsen.generator.OnThreads {:f #{:nemesis}, :gen nil})
                 (jepsen.generator.OnThreads
                  {:f #object[clojure.core$complement$fn__5686
                              "0x4c04188"
                              "clojure.core$complement$fn__5686@4c04188"],
                   :gen (jepsen.generator.Stagger
                         {:dt 20000000,
                          :next-time nil,
                          :gen ({:type :invoke, :f :txn, :value [[:append 9 1]]}
                                {:type :invoke,
                                 :f :txn,
                                 :value [[:append 9 2] [:r 5 nil] [:r 7 nil]]}
                                {:type :invoke,
                                 :f :txn,
                                 :value [[:append 7 1]
                                         [:append 9 3]
                                         [:append 9 4]]}
                                {:type :invoke,
                                 :f :txn,
                                 :value [[:r 9 nil]
                                         [:r 6 nil]
                                         [:r 8 nil]
                                         [:r 8 nil]]}
                                {:type :invoke,
                                 :f :txn,
                                 :value [[:r 8 nil]
                                         [:r 9 nil]
                                         [:r 9 nil]
                                         [:r 9 nil]]}
                                {:type :invoke,
                                 :f :txn,
                                 :value [[:append 9 5]
                                         [:append 7 2]
                                         [:append 9 6]
                                         [:append 9 7]]}
                                {:type :invoke,
                                 :f :txn,
                                 :value [[:append 7 3] [:r 8 nil] [:r 8 nil]]}
                                {:type :invoke, :f :txn, :value [[:append 9 8]]}
                                ...)})})]})})
 :log-net-recv false
 :os
 #object[maelstrom.net$jepsen_os$reify__15686
         "0x32441ed4"
         "maelstrom.net$jepsen_os$reify__15686@32441ed4"]
 :time-limit 5
 :workload :txn-list-append
 :consistency-models [:strict-serializable]
 :topology :grid}
 
INFO [2022-09-01 17:42:45,206] jepsen node n0 - maelstrom.net Starting Maelstrom network
INFO [2022-09-01 17:42:45,208] jepsen test runner - jepsen.db Tearing down DB
INFO [2022-09-01 17:42:45,210] jepsen test runner - jepsen.db Setting up DB
INFO [2022-09-01 17:42:45,212] jepsen node n0 - maelstrom.service Starting services: (lin-kv lin-tso lww-kv seq-kv)
INFO [2022-09-01 17:42:45,214] jepsen node n0 - maelstrom.db Setting up n0
INFO [2022-09-01 17:42:45,215] jepsen node n0 - maelstrom.process launching /usr/local/bin/persistcli ["maelstrom" "--blob-uri=mem://" "--consensus-uri=mem://"]
INFO [2022-09-01 17:42:45,244] jepsen test runner - jepsen.core Relative time begins now
INFO [2022-09-01 17:42:45,256] jepsen worker 0 - jepsen.util 0	:invoke	:txn	[[:append 9 1]]
INFO [2022-09-01 17:42:45,257] jepsen worker 1 - jepsen.util 1	:invoke	:txn	[[:append 9 2] [:r 5 nil] [:r 7 nil]]
INFO [2022-09-01 17:42:50,260] jepsen worker 1 - jepsen.util 1	:info	:txn	[[:append 9 2] [:r 5 nil] [:r 7 nil]]	:net-timeout
INFO [2022-09-01 17:42:50,260] jepsen worker 0 - jepsen.util 0	:info	:txn	[[:append 9 1]]	:net-timeout
INFO [2022-09-01 17:42:50,262] jepsen test runner - jepsen.core Run complete, writing
INFO [2022-09-01 17:42:50,277] jepsen node n0 - maelstrom.db Tearing down n0
INFO [2022-09-01 17:42:52,247] jepsen node n0 - maelstrom.net Shutting down Maelstrom network
INFO [2022-09-01 17:42:52,248] jepsen test runner - jepsen.core Analyzing...
INFO [2022-09-01 17:42:52,389] jepsen test runner - jepsen.core Analysis complete
INFO [2022-09-01 17:42:52,401] jepsen results - jepsen.store Wrote /store/txn-list-append/20220901T174242.000Z/results.edn
INFO [2022-09-01 17:42:52,408] jepsen test runner - jepsen.core {:perf {:latency-graph {:valid? true},
        :rate-graph {:valid? true},
        :valid? true},
 :timeline {:valid? true},
 :exceptions {:valid? true},
 :stats {:valid? false,
         :count 2,
         :ok-count 0,
         :fail-count 0,
         :info-count 2,
         :by-f {:txn {:valid? false,
                      :count 2,
                      :ok-count 0,
                      :fail-count 0,
                      :info-count 2}}},
 :net {:all {:send-count 6,
             :recv-count 6,
             :msg-count 6,
             :msgs-per-op 3.0},
       :clients {:send-count 4, :recv-count 4, :msg-count 4},
       :servers {:send-count 2,
                 :recv-count 2,
                 :msg-count 2,
                 :msgs-per-op 1.0},
       :valid? true},
 :workload {:valid? :unknown,
            :anomaly-types (:empty-transaction-graph),
            :anomalies {:empty-transaction-graph true},
            :not #{},
            :also-not #{}},
 :valid? false}
 
Analysis invalid! (?????? ???
mzcompose: test case workflow-default failed: running docker compose failed (exit status 1)
```

**Comment by aljoscha** [comment]
@danhhz and/or @pH14 could you please take a look?

**Comment by danhhz** [comment]
okay so we're getting a failure because jepsen calls any analysis that _only_ has timeouts (no successful reqs) invalid and that's what's happening here ```INFO [2022-09-01 17:42:45,256] jepsen worker 0 - jepsen.util 0	:invoke	:txn	[[:append 9 1]]
INFO [2022-09-01 17:42:45,257] jepsen worker 1 - jepsen.util 1	:invoke	:txn	[[:append 9 2] [:r 5 nil] [:r 7 nil]]
INFO [2022-09-01 17:42:50,260] jepsen worker 1 - jepsen.util 1	:info	:txn	[[:append 9 2] [:r 5 nil] [:r 7 nil]]	:net-timeout
INFO [2022-09-01 17:42:50,260] jepsen worker 0 - jepsen.util 0	:info	:txn	[[:append 9 1]]	:net-timeout```

this CI job is meant to bundle the maelstrom store directory up as an buildkite artifact for download, but it looks like somehow that bit got lost in the back-and-forth benesch and I did when getting the CI job set up. I'm gonna see if I can dig that up, because it'll be helpful in the future. I have a feeling that node logs would immediately pinpoint the problem here

I tried running this configuration locally for an extra long time but it passed. I think given that no requests are served in a repro, it's probably better to focus on very short runs but many of them

**Comment by danhhz** [comment]
For any future adventurers, I'm very interested in additional flakes that are happen on any commit including #14593, which should help immensely in debugging this. Please post here and ping me.

**Comment by philip-stoev** [comment]
@danhhz flake from yesterday https://buildkite.com/materialize/tests/builds/41061#018318cf-b054-46e9-bc25-e21ad358a083

**Comment by danhhz** [comment]
fix is up in #14745
